libraries/cms/Provider/ConfigurationProvider.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * @package     Joomla.Libraries
+ * @subpackage  Service
+ *
+ * @copyright   Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Cms\Provider;
+
+defined('JPATH_PLATFORM') or die;
+
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Registry\Registry;
+
+/**
+ * The CMS registry service provider which sets a global config object.
+ *
+ * @since  4.0
+ */
+class ConfigurationProvider implements ServiceProviderInterface
+{
+	public function register(Container $container)
+	{
+		// Getting a registry object
+		$container->set('Registry', array($this, 'getRegistry'));
```

Comment:
We don't need to add a Registry object with this key to the DIC.  Since Registry objects are pretty generic, we don't need some sort of wrapper around `new Registry` I think.  Configured and persistent Registry objects are OK to do this with.